File: src/codes/god-object/py/uncode.py
class OrderManager:
    def __init__(self):
        self.orders = []
        self.user_email = ""
        self.discount_type = "none"

    def set_user(self, email: str):
        self.user_email = email

    def add_order(self, product_id: str):
        self.orders.append(product_id)

    def apply_discount(self, discount_type: str):
        self.discount_type = discount_type

    def calculate_total(self) -> float:
        base_price = len(self.orders) * 1000
        if self.discount_type == "student":
            return base_price * 0.8
        elif self.discount_type == "member":
            return base_price * 0.9
        return base_price

    def confirm_order(self):
        # 1. 在庫調整
        print("在庫を減らす")

        # 2. メール通知
        print(f"メールを {self.user_email} に送信")

        # 3. ログ出力
        print("注文完了ログ")

File: src/codes/god-object/ts/uncode.ts
// Godクラス：責務が全部入り
class OrderManager {
  private orders: string[] = [];
  private userEmail: string = "";
  private discountType: string = "none";

  setUser(email: string) {
    this.userEmail = email;
  }

  addOrder(productId: string) {
    this.orders.push(productId);
  }

  applyDiscount(type: string) {
    this.discountType = type;
  }

  calculateTotal(): number {
    let basePrice = this.orders.length * 1000;
    if (this.discountType === "student") {
      return basePrice * 0.8;
    } else if (this.discountType === "member") {
      return basePrice * 0.9;
    }
    return basePrice;
  }

  confirmOrder() {
    // 1. 在庫調整
    console.log("在庫を減らす");

    // 2. メール通知
    console.log(`メールを ${this.userEmail} に送信`);

    // 3. ログ出力
    console.log("注文完了ログ");
  }
}
